flaml/autogen/oai/completion.py

```diff
@@ -649,15 +649,23 @@ def _construct_params(cls, data_instance, config, prompt=None, messages=None):
             if messages is None:
                 raise ValueError("Either prompt or messages should be in config for chat models.")
         if prompt is None:
-            params["messages"] = [
-                {
-                    "role": m["role"],
-                    "content": m["content"].format(**data_instance)
-                    if isinstance(m["content"], str)
-                    else m["content"](data_instance),
-                }
-                for m in messages
-            ]
+            params["messages"] = []
+            for m in messages:
+                if isinstance(m["content"], str):
+                    try:
+                        # try to format the message with the data instance
+                        content = m["content"].format(**data_instance)
+                    except Exception:
+                        # if it fails, use the raw message
```

Comment:
Add a warning message here to remind that the program failed to format the message with the data instance and thus raw message is being used?

Reply:
There might be a long sequence of texts and will be a lot of warning. We should assume that when use inputs messages instead of prompt, these message are less likely containing formatted strings. We can add comments to the doc to explain.